pkg/endpoint/public_ip_watcher.go

```diff
@@ -0,0 +1,117 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoint
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	submv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	v1 "github.com/submariner-io/submariner/pkg/client/clientset/versioned/typed/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/types"
+	"github.com/submariner-io/submariner/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog"
+)
+
+type PublicIPWatcherConfig struct {
+	SubmSpec      *types.SubmarinerSpecification
+	Interval      time.Duration
+	K8sClient     kubernetes.Interface
+	Endpoints     v1.EndpointInterface
+	LocalEndpoint types.SubmarinerEndpoint
+}
+
+type PublicIPWatcher struct {
+	config PublicIPWatcherConfig
+}
+
+const DefaultMonitorInterval = 20 * time.Second
+
+func NewPublicIPWatcher(config *PublicIPWatcherConfig) (*PublicIPWatcher, error) {
+	controller := &PublicIPWatcher{
+		config: *config,
+	}
+
+	if config.Interval == 0 {
+		config.Interval = DefaultMonitorInterval
```

Comment:
We need to set it in the copy instead of the passed instance.

```suggestion
	if controller.config.Interval == 0 {
		controller.config.Interval = DefaultMonitorInterval
```